CMakeLists.txt

```diff
@@ -63,6 +64,11 @@ if(CMAKE_BUILD_TYPE MATCHES "Debug")
     xcheck_add_c_compiler_flag(-fno-omit-frame-pointer)
 endif()
 
+check_struct_has_member("struct tm" tm_gmtoff "time.h" HAVE_TM_GMTOFF LANGUAGE C)
+if(HAVE_TM_GMTOFF)
+    add_compile_definitions(HAVE_TM_GMTOFF)
+endif()
```

Comment:
I don't think this is something we want to auto-detect but rather have it as a build option that one opts into explicitly.

Maybe it's better to incorporate the tzdata directly because that gives a consistent experience everywhere (been looking into that) but that doesn't have to hold up this PR.

Reply:
An explicit option would be fine to me as well, but I'm curious why we wouldn't want to leverage CMakes detection as a default?

Reply:
IMHO it would be best to set the define for the special case, so the build just works if you grab the bunch of files and create a simple make file.

Reply:
One reason is that we'd have to mimic it if we added e.g. a meson or autotools build script. Another is that it affects runtime semantics; that should be a conscious decision, not something the build system does behind your back.

Reply:
I'm not sure if I buy it.
* If other build systems are implemented, they may or may not have equivalent features. Doesn't seem like a requirement to me.
* Regarding runtime semantics, we're already at the mercy of the behavior of the libc implementation. Like you said, for total control, we should incorporate tzdata directly. What I'm going for here is a "best effort" with room for improvement.

But anyways, I'm also fine with removing the `check_struct_has_member` if that's what gets this PR over the finish line. Just let me know.

Reply:
The kind of auto-detection I could live with it are`#ifdef __NEWLIB__` guards in quickjs.c, assuming it's actually possible to feature-detect newlib at compile time<sup>1</sup>. Otherwise I'd like to see an explicit build flag.

<sup>1</sup> I'm fairly sure `#if defined(_NEWLIB_STDIO_H)` would work although it's maybe not the best way

Reply:
+1